bin/rule.ml

```diff
@@ -0,0 +1,69 @@
+(*
+ * Copyright (c) 2018 Thomas Gazagnaire <thomas@gazagnaire.org>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *)
+
+let src = Logs.Src.create "cram.rule"
+
+module Log = (val Logs.src_log src : Logs.LOG)
+
+let print_rule ~md_file ~ml_file =
+  let pct = '%' in
+  Fmt.pr
+    "\
+(alias\n\
+\ (name   runtest)\n\
+\ (deps   (:x %s) (:y %s.expected)\n\
+\         (:a %s) (:b %s.expected) (package mdx))\n\
+\ (action (progn\n\
+\           (run mdx test %c{x})\n\
+\           (diff? %c{y} %c{x}.corrected)\n\
+\           (diff? %c{b} %c{a}.corrected))))\n\n"
```

Comment:
I would have expected to have a list of `ml` files here, and have one rule per `md` file instead of one rule per block.  I am not sure that's even work properly :-)